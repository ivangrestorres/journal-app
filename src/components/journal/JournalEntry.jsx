import React from "react";

export const JournalEntry = () => {
    return (
        <div className={"journal_entry"}>
            <div
                className="journal_entry-picture"
                style={{
                    backgroundSize: "cover",
                    backgroundImage:
                        "url(https://st3.depositphotos.com/1813774/19042/i/600/depositphotos_190424116-stock-photo-historical-medieval-city-along-the.jpg)",
                }}
            ></div>
            <div className={"journal_entry-body"}>
                <p className={"journal_entry-title"}>Un nuevo dia</p>
                <p className={"journal_entry-content"}>
                    Est sint officia fugiat irure anim magna aliquip sit qui
                    mollit enim nisi occaecat anim. Ut duis aute mollit culpa
                    deserunt eiusmod incididunt aute cillum officia non. Aute
                    nisi aute anim in dolor et anim eu. Officia duis dolore
                    pariatur ut elit cupidatat dolor nostrud.
                </p>
            </div>
            <div className={"journal_entry-date-box"}>
                <span>Monday</span>
                <h4>28</h4>
            </div>
        </div>
    );
};
