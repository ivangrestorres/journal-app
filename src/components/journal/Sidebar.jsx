import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startLogout } from "../../actions/auth";
import { addNote } from "../../actions/notes";
import { JournalEntries } from "./JournalEntries";

export const Sidebar = () => {
    const dispatch = useDispatch();
    const { name } = useSelector((state) => state.auth);

    const handleLogOut = () => {
        dispatch(startLogout());
    };

    const handleAddNewNote = () => {
        dispatch(addNote());
    };

    return (
        <aside className={"journal_sidebar"}>
            <div className={"journal_sidebar-navbar"}>
                <h4 className={"mt-1"}>
                    <i className={"far fa-user"}></i>
                    <span> {name}</span>
                </h4>
                <button onClick={handleLogOut} className={"btn"}>
                    Logout
                </button>
            </div>
            <div onClick={handleAddNewNote} className={"journal_new-entry"}>
                <i className="far fa-calendar-plus fa-3x"></i>
                <p className="mt-5">New entry</p>
            </div>
            <JournalEntries />
        </aside>
    );
};
