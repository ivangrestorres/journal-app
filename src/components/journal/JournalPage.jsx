import React from "react";
import { NoteScreen } from "../notes/NoteScreen";
// import { NothingSelected } from "./NothingSelected";
import { Sidebar } from "./Sidebar";

export const JournalPage = () => {
    return (
        <div className={"journal_main-content"}>
            <Sidebar />
            <main>
                {/* <NothingSelected /> */}
                <NoteScreen />
            </main>
        </div>
    );
};
