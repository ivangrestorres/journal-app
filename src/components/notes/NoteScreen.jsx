import React from "react";
import { useRef } from "react";
import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { activeNote, startDeleting } from "../../actions/notes";
import { useForm } from "../../hooks/useForm";
import { NotesAppBar } from "./NotesAppBar";

export const NoteScreen = () => {
    const dispatch = useDispatch();

    const { active: note } = useSelector((state) => state.notes);

    const [formValues, handleInputChange, reset] = useForm(note);
    const { title, body } = formValues;

    const activeId = useRef(note.id);

    useEffect(() => {
        if (note.id !== activeId.current) {
            reset(note);
            activeId.current = note.id;
        }
    }, [note, reset]);

    useEffect(() => {
        dispatch(activeNote(formValues.id, { ...formValues }));
    }, [formValues, dispatch]);

    const handleDelete = () => {
        dispatch(startDeleting(note.id));
    };

    return (
        <div className={"notes_main-content"}>
            <NotesAppBar />
            <div className={"notes_content"}>
                <input
                    type="text"
                    placeholder={"Some awesome title"}
                    className={"notes_title-input"}
                    name={"title"}
                    value={title}
                    onChange={handleInputChange}
                    autoComplete={"off"}
                />
                <textarea
                    placeholder={"What happened today"}
                    value={body}
                    name={"body"}
                    onChange={handleInputChange}
                    className={"notes_textarea"}
                ></textarea>
                {note.url && (
                    <div>
                        <img
                            src={note.url}
                            alt="imagen"
                            className={"notes_image"}
                        />
                    </div>
                )}
            </div>
            <button onClick={handleDelete} className={"btn btn-danger"}>
                Delete
            </button>
        </div>
    );
};
