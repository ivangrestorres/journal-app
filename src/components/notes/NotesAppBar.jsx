import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startSaveNote, startUploading } from "../../actions/notes";

export const NotesAppBar = () => {
    const dispatch = useDispatch();
    const { active: note } = useSelector((state) => state.notes);

    const handleSaveNote = () => {
        dispatch(startSaveNote(note));
    };

    const handleFileChange = ({ target }) => {
        const file = target.files[0];
        if (file) {
            dispatch(startUploading(file));
        }
    };

    const handlePicture = () => {
        document.querySelector("#fileSelector").click();
    };

    return (
        <div className={"notes_appbar"}>
            <span>28 de agosto de 2020</span>
            <input
                id={"fileSelector"}
                onChange={handleFileChange}
                type="file"
                style={{ display: "none" }}
            />
            <div>
                <button onClick={handlePicture} className={"btn"}>
                    Picture
                </button>
                <button onClick={handleSaveNote} className={"btn"}>
                    Save
                </button>
            </div>
        </div>
    );
};
