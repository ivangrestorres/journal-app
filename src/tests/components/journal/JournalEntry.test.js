import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom";

import { JournalEntry } from "../../../components/journal/JournalEntry";
import { activeNote } from "../../../actions/notes";

jest.mock("../../../actions/notes", () => ({
    activeNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
    id: 10,
    date: 0,
    title: "Hola",
    body: "Mundo",
};

describe("Pruebas <JournalEntry />", () => {
    beforeEach(() => {
        store = mockStore(initState);
        jest.clearAllMocks();
    });

    const wrapper = mount(
        <Provider store={store}>
            <JournalEntry {...note}></JournalEntry>
        </Provider>
    );
    test("debe de mostrar correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de activar la nota", () => {
        wrapper.find(".journal_entry").prop("onClick")();
        expect(activeNote).toHaveBeenCalled();
    });
});
