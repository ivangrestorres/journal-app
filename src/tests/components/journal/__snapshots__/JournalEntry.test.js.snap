// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <JournalEntry /> debe de mostrar correctamente 1`] = `
<div
  className="journal_entry animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal_entry-body"
  >
    <p
      className="journal_entry-title"
    >
      Hola
    </p>
    <p
      className="journal_entry-content"
    >
      Mundo
    </p>
  </div>
  <div
    className="journal_entry-date-box"
  >
    <span>
      Thu
    </span>
    <h4>
      1st
    </h4>
  </div>
</div>
`;
